packages/authentication/src/decorators/authenticate.decorator.ts

```diff
@@ -29,7 +29,7 @@ class AuthenticateClassDecoratorFactory extends ClassDecoratorFactory<
  * @param options - Additional options to configure the authentication.
  */
 export function authenticate(
-  strategyNameOrMetadata: string | AuthenticationMetadata,
+  strategyNameOrMetadata: string | string[] | AuthenticationMetadata,
```

Comment:
Let's improve the user experience here. I would like to see the following styles:

1. Allow one or more strategy names
```ts
@authenticate('my-strategy-1', 'my-strategy-2')
```
2. Allow one or more meta data objects
```
@authenticate(metaForStrategy1, metaForStrategy2)
```
3. Allow multiple applications of `@authenticate` on the same method: (nice-to-have)
```ts
@authenticate('my-strategy-1');
@authenticate(metaForMyStrategy1);
```

The signature can be:
```ts
export function authenticate(...strategies: (string | AuthenticationMetadata) []) { ... }
```
It allows one or more `string` or `AuthenticationMetadata`.

Please note that different strategy may have different metadata.

Reply:
I did look into that but this means that the changes will be breaking for sure since there are no more options in the decorator, e.g. this will fail now
```ts
it('can add authenticate metadata to target method with options', () => {
      class TestClass {
        @authenticate('my-strategy', {option1: 'value1', option2: 'value2'})
        whoAmI() {}
      }

      const metaData = getAuthenticateMetadata(TestClass, 'whoAmI');
      expect(metaData).to.eql({
        strategy: 'my-strategy',
        options: {option1: 'value1', option2: 'value2'},
      });
    });
```

One more thing I am not sure about is how the metadata for each strategy should be injected into a strategy since it is now an array of metadata object.

```ts
@inject(AuthenticationBindings.METADATA)
 private metadata: AuthenticationMetadata[], // this is now an array
```

The solution could be to iterate through the metadata array and find the according metadata based on the strategy name, maybe a helper function could be added or can this be done in the inject somehow?

Reply:
Great question.

For `@inject`, I don't think we can resolve it to the metadata for a given strategy name, which is unknown.

Maybe we have to use `AuthenticationMetadata[]` as the type and introduce a helper function to get `AuthenticationMetadata` by strategy name.